source/loader/layers/sanitizer/asan_interceptor.cpp

```diff
@@ -366,23 +363,40 @@ ur_result_t SanitizerInterceptor::enqueueMemSetShadow(
         uptr ShadowEnd =
             MemToShadow_CPU(DeviceInfo->ShadowOffset, Ptr + Size - 1);
 
-        const char Pattern[] = {(char)Value};
-        auto URes = context.urDdiTable.Enqueue.pfnUSMFill(
-            Queue, (void *)ShadowBegin, 1, Pattern, ShadowEnd - ShadowBegin + 1,
-            NumEventsInWaitList, EventsWaitList, Event);
+        // Poison shadow memory outside of asan runtime is not allowed, so we
+        // need to avoid memset's call from being intercepted.
+        static void *memset_ptr = []() {
+            void *handle = dlopen("libc.so.6", RTLD_LAZY);
```

Comment:
should we hardcode the version number?

Reply:
Makes sense. I changed it to `LIBC_SO` macro.